pkg/user/tx_client.go

```diff
@@ -379,40 +393,175 @@ func (client *TxClient) BroadcastTx(ctx context.Context, msgs []sdktypes.Msg, op
+	go func() {
+		// Wait for all goroutines in the group to complete.
+		// Assign groupErr within this goroutine, but handle it after the select.
+		groupErr := g.Wait()
+		if groupErr != nil && !errors.Is(groupErr, context.Canceled) {
+			// If there was a non-cancellation error from the group itself and
+			// finalErr wasn't already set by context timeout/cancellation,
+			// store it. We need to be careful about races if multiple goroutines
+			// returned non-nil, but errgroup should only return the first.
+			// Using a lock or atomic might be safer if we expected multiple
+			// actual errors returned from g.Go functions (which we don't here).
+			if finalErr == nil { // Avoid overwriting a context error
+				finalErr = groupErr
+			}
+		}
+		close(done)
+	}()
+
+	select {
+	case firstResp = <-respCh:
+		// Success from one connection. Immediately cancel other in-flight requests.
+		cancel()
+	case <-childCtx.Done(): // Parent context cancelled or deadline exceeded before first success
+		finalErr = childCtx.Err()
+		// If the context finished, ensure respCh is drained if a response arrived concurrently
+		select {
+		case firstResp = <-respCh:
+			cancel() // Also cancel if we got a response just as context finished
+		default:
+		}
+	case <-done: // All goroutines finished without success or context cancellation
+		// Proceed to error collection below
+	}
+
+	// Wait for the errgroup goroutine (which includes g.Wait()) to finish.
+	// This ensures all broadcast goroutines have completed and potentially sent
+	// errors to errCh before we try to close and read from it.
+	<-done
+
+	// If we fall through the select because <-done was triggered or context expired,
+	// we still need to potentially drain respCh in case a response arrived concurrently.
+	if firstResp == nil {
+		select {
+		case firstResp = <-respCh:
+			// A response arrived very late, just as g.Wait() finished.
+			successfulBroadcast.Do(cancel) // Ensure cancellation signal is sent only once
+		default:
+		}
 	}
-	if resp.TxResponse.Code != abci.CodeTypeOK {
-		broadcastTxErr := &BroadcastTxError{
-			TxHash:   resp.TxResponse.TxHash,
-			Code:     resp.TxResponse.Code,
-			ErrorLog: resp.TxResponse.RawLog,
+
+	// If we received a successful response (either from select or late drain)
+	if firstResp != nil {
+		successfulBroadcast.Do(func() {
+			// Ensure cancellation happens if we got a late success after <-done
+			cancel()
+			client.txTracker[firstResp.TxHash] = txInfo{
+				sequence:  client.signer.accounts[signer].Sequence(),
+				signer:    signer,
+				timestamp: time.Now(),
+			}
+
+			incErr := client.signer.IncrementSequence(signer)
+			if incErr != nil {
+				fmt.Printf("CRITICAL: Failed to increment sequence for signer %s after successful broadcast: %v\n", signer, incErr)
+				// TODO: Wrap the increment error and return it?
+				// finalErr = fmt.Errorf("increment sequencing failed after successful broadcast: %w", incErr)
+				// firstResp = nil
```

Comment:
[blocking] commented code is a red flag. If the commented code isn't applicable, can it be removed and included in a follow-up issue created to address this TODO?

Ref: https://kentcdodds.com/blog/please-dont-commit-commented-out-code